File: .taira-source-c80bf4d85a97/java/iroha_android/src/main/java/org/hyperledger/iroha/android/client/RamLfeReceiptVerifyRequest.java
package org.hyperledger.iroha.android.client;

import java.util.Collections;
import java.util.LinkedHashMap;
import java.util.Map;
import java.util.Objects;

/** Typed request wrapper for RAM-LFE receipt verification. */
public final class RamLfeReceiptVerifyRequest {
  private final Map<String, Object> receipt;
  private final String outputHex;

  public RamLfeReceiptVerifyRequest(final Map<String, Object> receipt, final String outputHex) {
    this.receipt =
        Collections.unmodifiableMap(new LinkedHashMap<>(Objects.requireNonNull(receipt, "receipt")));
    this.outputHex = outputHex;
  }

  public Map<String, Object> receipt() {
    return receipt;
  }

  public String outputHex() {
    return outputHex;
  }
}

File: kotlin/core-jvm/src/main/java/org/hyperledger/iroha/sdk/client/RamLfeReceiptVerifyRequest.kt
package org.hyperledger.iroha.sdk.client

/** Typed request wrapper for RAM-LFE receipt verification. */
class RamLfeReceiptVerifyRequest(
    receipt: Map<String, Any>,
    @JvmField val outputHex: String?,
) {
    @JvmField
    val receipt: Map<String, Any> = receipt.toMap()
}
